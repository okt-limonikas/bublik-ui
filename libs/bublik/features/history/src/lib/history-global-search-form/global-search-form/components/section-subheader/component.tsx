/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { FC } from 'react';

export interface FormSectionSubheaderProps {
	name: string;
}

export const FormSectionSubheader: FC<FormSectionSubheaderProps> = ({
	name
}) => {
	return (
		<div className="mb-2">
			<span className="text-[0.6875rem] leading-[0.875rem]">{name}</span>
		</div>
	);
};
