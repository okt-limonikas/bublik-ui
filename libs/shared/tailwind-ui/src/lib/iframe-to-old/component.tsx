/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, FC } from 'react';

import { cn } from '../utils';

export type IframeToOldBublikProps = ComponentPropsWithRef<'iframe'>;

export const IframeToOldBublik: FC<IframeToOldBublikProps> = (props) => {
	return (
		<iframe
			className={cn('w-full h-full border-none', props.className)}
			title="Old Bublik"
			{...props}
		/>
	);
};
