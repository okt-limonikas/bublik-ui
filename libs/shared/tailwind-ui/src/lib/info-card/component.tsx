/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, forwardRef } from 'react';

export interface InfoCardProps extends ComponentPropsWithRef<'div'> {
	header: string;
	message: string;
}

export const InfoCard = forwardRef<HTMLDivElement, InfoCardProps>(
	(props, ref) => {
		const { header, message } = props;

		return (
			<div
				className="flex items-center justify-center w-full h-screen"
				data-testid="info-card"
			>
				<div
					className="bg-white min-w-[600px] max-w-[768px] flex flex-col rounded-2xl shadow-2xl"
					ref={ref}
				>
					<div className="p-4 border-b border-border-primary">
						<h2>{header}</h2>
					</div>
					<div className="flex-1 p-4">
						<p>{message}</p>
					</div>
				</div>
			</div>
		);
	}
);
