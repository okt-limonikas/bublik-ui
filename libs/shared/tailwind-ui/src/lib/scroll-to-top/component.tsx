/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, FC, forwardRef } from 'react';
import { AnimatePresence, motion, Variants } from 'framer-motion';

import { useScrollToTopPage } from '@/shared/hooks';

import { Icon } from '../icon';

const variants: Variants = {
	visible: { y: 0, opacity: 1 },
	hidden: { y: 72, opacity: 0 }
};

export interface ScrollToTopProps extends ComponentPropsWithRef<'button'> {
	offset?: number;
}

export const ScrollToTopPage: FC<ScrollToTopProps> = forwardRef(
	({ offset = 0 }, ref) => {
		const { isVisible, scrollToTop } = useScrollToTopPage(offset);

		return (
			<AnimatePresence>
				{isVisible && (
					<motion.div
						variants={variants}
						initial="hidden"
						animate="visible"
						exit="hidden"
						transition={{ type: 'just' }}
						className="fixed z-40 bottom-4 right-4"
					>
						<button
							className="flex items-center gap-1 p-3 text-white rounded-full bg-primary transition-all text-[1rem] font-medium leading-[0.875rem] border-2 border-transparent hover:border-[#94b0ff]"
							onClick={scrollToTop}
							data-testid="scroll-to-top"
							aria-label="Scroll to top"
							ref={ref}
						>
							<Icon name="ArrowShortTop" className="grid place-items-center" />
						</button>
					</motion.div>
				)}
			</AnimatePresence>
		);
	}
);
