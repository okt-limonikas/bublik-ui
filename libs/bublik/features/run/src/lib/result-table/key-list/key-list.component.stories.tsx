/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import type { Meta, StoryObj } from '@storybook/react';
import { KeyList } from './key-list.component';
import { withBackground } from '@/shared/tailwind-ui';

const Story: Meta<typeof KeyList> = {
	component: KeyList,
	title: 'run/Key List',
	decorators: [withBackground]
};
export default Story;

type Story = StoryObj<typeof KeyList>;
export const Primary: Story = {
	args: {
		items: [
			{ name: 'NO-GENEVA', url: 'https://localhost:4200/v1' },
			{ name: 'NO-GENEVA', url: '' }
		]
	}
};

export const Url: Story = {
	args: {
		items: [{ name: 'NO-GENEVA', url: 'https://localchost:4200/1' }]
	}
};

export const Button: Story = {
	args: {
		items: [{ name: 'NO-GENEVA' }]
	}
};
