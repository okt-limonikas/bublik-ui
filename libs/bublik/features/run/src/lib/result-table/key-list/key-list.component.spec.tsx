/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';

import { KeyList } from './key-list.component';

describe('KeyList', () => {
	it('should render correctly anchor with url', () => {
		const items = [{ name: 'Item 1', url: 'https://example.com/1' }];

		const { asFragment } = render(<KeyList items={items} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render button when no url', () => {
		const items = [{ name: 'Item 1' }];

		const { asFragment } = render(<KeyList items={items} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render multiple items with url and without', () => {
		const items = [
			{ name: 'Item 1', url: 'https://example.com/1' },
			{ name: 'Item 2' },
			{ name: 'Item 3', url: 'https://example.com/3' }
		];

		const { asFragment } = render(<KeyList items={items} />);

		expect(asFragment()).toMatchSnapshot();
	});
});
