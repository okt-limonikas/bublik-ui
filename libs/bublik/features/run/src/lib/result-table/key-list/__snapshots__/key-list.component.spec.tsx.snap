// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KeyList > should render button when no url 1`] = `
<DocumentFragment>
  <ul
    class="flex flex-wrap gap-1"
  >
    <li>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary"
      >
        Item 1
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`KeyList > should render correctly anchor with url 1`] = `
<DocumentFragment>
  <ul
    class="flex flex-wrap gap-1"
  >
    <li>
      <a
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary underline text-primary"
        href="https://example.com/1"
        rel="noreferrer"
        target="_blank"
      >
        Item 1
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`KeyList > should render multiple items with url and without 1`] = `
<DocumentFragment>
  <ul
    class="flex flex-wrap gap-1"
  >
    <li>
      <a
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary underline text-primary"
        href="https://example.com/1"
        rel="noreferrer"
        target="_blank"
      >
        Item 1
      </a>
    </li>
    <li>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary"
      >
        Item 2
      </div>
    </li>
    <li>
      <a
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary underline text-primary"
        href="https://example.com/3"
        rel="noreferrer"
        target="_blank"
      >
        Item 3
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
